internal/controller/controllers/agent_controller.go

```diff
@@ -871,7 +871,12 @@ func (r *AgentReconciler) updateIfNeeded(ctx context.Context, log logrus.FieldLo
 		params.HostUpdateParams.MachineConfigPoolName = &spec.MachineConfigPool
 	}
 
-	if spec.Role != "" && spec.Role != internalHost.Role {
+	isUpdateRoleAllowed := func() bool {
+		//may update role to auto-assign only at initialization time
+		//otherwise, let the BE do the assignment to prevent races
+		return internalHost.Role == "" || spec.Role != models.HostRoleAutoAssign
```

Comment:
So if I set a role in CIM then I can never get it back to auto-assign?  That's awkward.

Reply:
Wouldn't reconciling all of the cluster's hosts before starting the installation fix the issue?